app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class RelationshipsController < ApplicationController
+  def create
+    followee = User.find_by(id: params.dig(:relationship, :followee_id))
+    if followee
```

Comment:
```suggestion
    if current_user.following_relationships.create(followee: followee)
```

でいいかもしれませんね。

- followeeがnilだったら、必須チェックのバリデーションに引っかかる
- すでにfolloweeをフォロー中だったら、ユニーク制約のバリデーションに引っかかる
- 将来的に新たなバリデーションが追加されたら、それはそれでエラーのときに引っかかる

という形になるので都合が良さそうです。
ただ、画面には「フォローに失敗しました」的なメッセージが出ていることが望ましいですが。

Reply:
質問ですm

> - followeeがnilだったら、必須チェックのバリデーションに引っかかる
> - 将来的に新たなバリデーションが追加されたら、それはそれでエラーのときに引っかかる

上記の2パターンについては、「フォローボタンを押したのに、フォローできなかった」状態になるのでなんらかのエラーが出る必要があると思うのですが、

こちら
> - すでにfolloweeをフォロー中だったら、ユニーク制約のバリデーションに引っかかる

のパターンについては、「フォローボタンを押したら既にフォロー済みだった＝結果的にはフォローできている」状態なので、ユーザー的には特に何もする必要がないため、エラーメッセージを出さないほうが親切ではないかな？と思ったのですが、いかがでしょうか🤔

現状は、

- followeeがnilだったら、必須チェックのバリデーションに引っかかる
   - 「対象のユーザが見つかりませんでした」のエラーメッセージを出す
- すでにfolloweeをフォロー中だったら、ユニーク制約のバリデーションに引っかかる
  - エラーメッセージを出さない
- 将来的に新たなバリデーションが追加されたら、それはそれでエラーのときに引っかかる
  - エラー画面が出ていい（https://github.com/mayumonj/fjord-books_app/pull/5#discussion_r939467638 のコメントを受けて、例外処理は削除）

となるように作ったつもりでした。
違和感ありますでしょうか？